packages/orm/src/relations/CustomCollection.ts

```diff
@@ -15,6 +15,7 @@ export type CustomCollectionOpts<T extends Entity, U extends Entity> = {
   remove?: (entity: T, other: U) => void;
   /** Whether the reference is loaded, even w/o an explicit `.load` call, i.e. for DeepNew test instances. */
   isLoaded: () => boolean;
+  loadHint?: ((entity: T) => LoadHint<T>) | LoadHint<T>;
```

Comment:
Maybe jsdoc when/why CustomCollections/References should provide the `loadHint`? Like as a user I will probably be wondering "should I always implement this? What happens / what breaks / doesn't work if I don't?"